packages/auth/src/utils/passkey/registerPasskey.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { PasskeyCreateOptionsJson, PasskeyCreateResult } from './types';
+import {
+	deserializeJsonToPkcCreationOptions,
+	serializePkcToJson,
+} from './serde';
+import { PasskeyErrorCode, assertPasskeyError } from './errors';
+
+/**
+ * Registers a new passkey for user
+ * @param input - PasskeyCreateOptions
+ * @returns serialized PasskeyCreateResult
+ */
+export const registerPasskey = async (input: PasskeyCreateOptionsJson) => {
+	const passkeyCreationOptions = deserializeJsonToPkcCreationOptions(input);
+
+	const credential = (await navigator.credentials.create({
+		publicKey: passkeyCreationOptions,
+	})) as PasskeyCreateResult | null;
```

Comment:
Nit, can `navigator.credentials` be unavailable in http? Maybe this will help throw more useful error message:
```suggestion
	const credential = (await navigator.credentials?.create({
		publicKey: passkeyCreationOptions,
	})) as PasskeyCreateResult | null;
```

Also, we need to mention to check security context in the recovery suggestion.

Reply:
Nice catch. I will add a specific code path to check for platform support something like `getIsPasskeySupported` that will throw if unavailable and add a helpful recovery message 👍.

That will also be useful to have to check for platform support on native when we get there...

Reply:
Updated.